### PR TITLE
Feature/alcs 891

### PR DESCRIPTION
**Pull request by mhuseinov**
- search support for NOI, APP, COV, PLAN

### CODE REVIEW

File: alcs-frontend/src/app/services/search/search.service.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  private baseUrl = `${environment.apiUrl}/search`;
+
+  constructor(private http: HttpClient, private toastService: ToastService) {}
+
+  async fetch(searchTerm: string) {
+    try {
+      return await firstValueFrom(this.http.get<SearchResultDto[]>(`${this.baseUrl}/${searchTerm}`));
+    } catch (e) {
+      console.error(e);
+      this.toastService.showErrorToast(`Search filed for ${searchTerm}`);
```
Comment (dhaselhan) [inline alcs-frontend/src/app/services/search/search.service.ts:21]: Typo: filed -> failed

File: alcs-frontend/src/app/shared/header/search/search.component.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        this.searchService
+          .fetch(searchText)
+          .then((result) => (this.searchResults = this.mapSearchResults(result ?? [])));
+      }
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.searchResults = [];
```
Comment (dhaselhan) [inline alcs-frontend/src/app/shared/header/search/search.component.ts:54]: Does this do anything?

File: alcs-frontend/src/app/shared/header/search/search.component.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+      return {
+        title: `${e.fileNumber} ${e.applicant ?? ''}`,
+        class: classType,
+        type: e.type,
+        label: label,
+        government: e.localGovernmentName,
+        referenceId: e.referenceId,
+        board: e.boardCode,
+      } as SearchResult;
```
Comment (dhaselhan) [inline alcs-frontend/src/app/shared/header/search/search.component.ts:90]: Why are we casting? Can we do this cleaner?

File: alcs-frontend/src/app/shared/header/search/search.component.ts (hunk 2 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/shared/header/search/search.component.ts]: updated

File: alcs-frontend/src/app/shared/header/search/search.component.ts (hunk 3 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/shared/header/search/search.component.ts]: done

File: services/apps/alcs/src/alcs/alcs.module.ts [hunk 4]
```diff
@@ -35,6 +36,7 @@ import { StaffJournalModule } from './staff-journal/staff-journal.module';
     NoticeOfIntentModule,
     StaffJournalModule,
     NoticeOfIntentDecisionModule,
+    SearchModule,
```
Comment (dhaselhan) [inline services/apps/alcs/src/alcs/alcs.module.ts:39]: Looks like SearchModule is imported twice

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: alcs-frontend/src/app/shared/shared.module.ts [hunk 5]
```diff
@@ -82,6 +82,8 @@ import { ExtensionsDatepickerFormatter } from './utils/extensions-datepicker-for
     StaffJournalNoteComponent,
     StaffJournalNoteInputComponent,
     InlineReviewOutcomeComponent,
+    // TODO remove this
```
[the remark this answers is not in the document]
Comment (mhuseinov) [reply alcs-frontend/src/app/shared/shared.module.ts]: yep, removed

**Pull request merged**